pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -90,11 +90,14 @@ rec {
       installPhase =
         with builtins;
         let
+          # set toolchains via environment properties in wrapper and in gradle.properties.
+          # See https://docs.gradle.org/current/userguide/toolchains.html#sec:custom_loc
           toolchain = rec {
             prefix = x: "JAVA_TOOLCHAIN_NIX_${toString x}";
             varDefs = (lib.imap0 (i: x: "${prefix i} ${x}") javaToolchains);
             varNames = lib.imap0 (i: x: prefix i) javaToolchains;
             property = " -Porg.gradle.java.installations.fromEnv='${concatStringsSep "," varNames}'";
```

Comment:
Why are we keeping toolchains from env? Can't we drop this way of configuring toolchains now?

Reply:
The encoding as `JAVA_TOOLCHAIN_NIX_` does not seem to be documented as a user-facing aspect of the gradle wrapper.

We still want to set `JAVA_HOME` in the wrapper, right?